src/hotspot/share/asm/assembler.hpp

```diff
@@ -241,6 +241,20 @@ class AbstractAssembler : public ResourceObj  {
     }
   };
   friend class InstructionMark;
+
+  // count size of instructions which are skipped from inline heuristics
+  class InlineSkippedInstructionsCounter: public StackObj {
+   private:
+    AbstractAssembler* _assm;
```

Comment:
May be you should record `_assm->_code_section` instead and directly call `_code_section->register_skipped(_start)` instead of adding `AbstractAssembler:: register_skipped(size)`. With `CodeSection` code:
```
   register_skipped(address start) {
      _skipped_instructions_size += (_end - start);
   }
```

Reply:
I could do it as you wish. Please, tell me which variant is preferable for you.
PS But personally, I don't like putting actions inside destructors due to implicitness/non-visibility.

Reply:
There are a lot of classes that do that (the scoped accumulator pattern) already in the JVM. Let's move it do the destructor.

Reply:
done